Library/Homebrew/livecheck/strategy/apache.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Homebrew::Livecheck::Strategy::Apache
+  module_function
+
+  def match?(url)
+    %r{www\.apache\.org/dyn}.match?(url)
+  end
+
+  def find_versions(url, regex = nil)
+    path, prefix, suffix = url.match(%r{path=(.+?)/([^/]*?)\d+(?:\.\d+)+(/|[^/]*)})[1, 3]
```

Comment:
This regex (and most of these strategies) is a bit inpeneratrable. You may want to use the multiline regex format, add comments, assign to a constant and explain what `1` and `3` refer to using named match groups instead.

Reply:
I've included this issue in the new PR for strategy migration (in the PR description), so I'll resolve it here.